### PR TITLE
Framework: Implement skipped test case state

### PR DESCRIPTION
**Pull request by ader1990**
Implements feature request: https://github.com/LIS/LISAv2/issues/355

### CODE REVIEW

File: TestControllers/TestController.psm1 [hunk 1]
```diff
@@ -168,9 +169,13 @@ Class TestController
 		# XML secrets, used in Upload-TestResultToDatabase
 		Set-Variable -Name XmlSecrets -Value $this.XmlSecrets -Scope Global -Force
 		# Test results
-		Set-Variable -Name resultPass -Value "PASS" -Scope Global
-		Set-Variable -Name resultFail -Value "FAIL" -Scope Global
-		Set-Variable -Name resultAborted -Value "ABORTED" -Scope Global
+		$passResult = "PASS"
+		$skippedResult = "SKIPPED"
+		Set-Variable -Name ResultPassed  -Value $passResult -Scope Global
```
Comment (iamshital) [inline TestControllers/TestController.psm1]: Can you just put "PASS" and remove $passResult = "PASS"? same for skipped.

File: TestControllers/TestController.psm1 [hunk 2]
```diff
@@ -317,8 +322,8 @@ Class TestController
 					-TestLocation $this.TestLocation -TestProvider $this.TestProvider `
 					-Timeout $timeout -GlobalConfig $this.GlobalConfig
 				# Some cases returns a string, some returns a result object
-				if ($testResult.TestResult) {
-					$currentTestResult = $testResult
+				if ($testResult) {
+					$currentTestResult.TestResult = $testResult
```
Comment (iamshital) [inline TestControllers/TestController.psm1]: This is wrong. It should be `$currentTestResult = $testResult`
This has resulted in #709

File: TestControllers/TestController.psm1 [hunk 3]
```diff
@@ -342,8 +347,8 @@ Class TestController
 				-CurrentTestResult $currentTestResult -enableTelemetry $this.EnableTelemetry
 		}
 
-		$collectDetailLogs = $currentTestResult.TestResult -ne "PASS" -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
-		$doRemoveFiles = $currentTestResult.TestResult -eq "PASS" -and !$this.DoNotDeleteVMs -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
+		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
```
Comment (iamshital) [inline TestControllers/TestController.psm1]: Why this change? `$currentTestResult.TestResult -ne "$Global:ResultPassed"` should work.

File: TestControllers/TestController.psm1 [hunk 4]
```diff
@@ -342,8 +347,8 @@ Class TestController
...
+		$doRemoveFiles = $this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.DoNotDeleteVMs -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
```
Comment (iamshital) [inline TestControllers/TestController.psm1]: why not `$currentTestResult.TestResult -eq "$Global:ResultPassed"`?

File: TestControllers/TestController.psm1 (hunk 2 above)
Comment (iamshital) [reply TestControllers/TestController.psm1]: @ader1990 see the `job-cloudtest-auto/26329` I ran against your code.
It is also adding extra space e.g. `" PASSED"` in Telemetry SQL query

File: TestControllers/TestController.psm1 [hunk 5]
```diff
@@ -317,8 +322,8 @@ Class TestController
 					-TestLocation $this.TestLocation -TestProvider $this.TestProvider `
 					-Timeout $timeout -GlobalConfig $this.GlobalConfig
 				# Some cases returns a string, some returns a result object
-				if ($testResult.TestResult) {
-					$currentTestResult = $testResult
+				if ($testResult) {
```
Comment (iamshital) [inline TestControllers/TestController.psm1]: No need to change this.

**Comment by iamshital** [comment]
@ader1990 Adding "DO-NOT-MERGE" until #709 is solved. Please remove this flag once you fix the issue.

File: TestControllers/TestController.psm1 (hunk 4 above)
Comment (ader1990) [reply TestControllers/TestController.psm1]: SKIPPED is equivalent with PASS from the point of view of cleaning workspace after test has been run.

File: TestControllers/TestController.psm1 (hunk 3 above)
Comment (ader1990) [reply TestControllers/TestController.psm1]: SKIPPED is equivalent with PASS from the point of view of cleaning workspace after test has been run.

File: TestControllers/TestController.psm1 (hunk 1 above)
Comment (ader1990) [reply TestControllers/TestController.psm1]: SKIPPED is equivalent with PASS from the point of view of cleaning workspace after a test has been run. the state skipped and pass should be handled in the same way.

File: TestControllers/TestController.psm1 (hunk 5 above)
Comment (ader1990) [reply TestControllers/TestController.psm1]: Run-TestScript returns a string with the state if a bash script is the test script.
I need to further check in order to make sure that Run-TestScript does not return different kind of results depending on the test script being bash/python or powershell.

File: TestControllers/TestController.psm1 (hunk 2 above)
Comment (ader1990) [reply TestControllers/TestController.psm1]: ditto.

Will check the SQL problem and fix it.

**Comment by ader1990** [comment]
@iamshital can you check if the new fixes are alright? thanks.

**Comment by ader1990** [comment]
Execution tested:

```
[azure] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[azure] Total Test Cases      : 3 (2 Passed, 0 Failed, 0 Aborted, 1 Skipped)
[azure] Total Time (dd:hh:mm) : 0:0:11
[azure] 
[azure]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[azure] ---------------------------------------------------------------------------------------------------------------------
[azure]     1 BVT-LIS-DRIVER-VERSION-CHECK                                                   SKIPPED                 0.87 
[azure]     2 BVT-CORE-VERIFY-LIS-MODULES                                                       PASS                 0.77 
[azure]     3 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 3.27 
[azure] 	FirstBoot : PASS 
[azure] 	FirstBoot : Call Trace Verification : PASS 
[azure] 	Reboot : PASS 
[azure] 	Reboot : Call Trace Verification : PASS 
[azure] 
[azure] 
```

**Comment by ader1990** [comment]
Execution tested on HyperV:

```
[hyperv] Test Category         : BVT
[hyperv] Total Test Cases      : 9 (9 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[hyperv] Total Time (dd:hh:mm) : 0:0:29
[hyperv] 
[hyperv]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[hyperv] ---------------------------------------------------------------------------------------------------------------------
[hyperv]     1 BVT-CORE-RELOAD-MODULES-SMP                                                       PASS                11.58 
[hyperv]     2 BVT-CORE-TIME-CLOCKEVENT-SMP                                                      PASS                 1.89 
[hyperv]     3 BVT-CORE-TIMESYNC-NTP                                                             PASS                 0.79 
[hyperv]     4 BVT-CORE-VERIFY-LIS-MODULES                                                       PASS                 1.03 
[hyperv]     5 BVT-CORE-VERIFY-LIS-MODULES-VERSION                                               PASS                 1.03 
[hyperv]     6 BVT-CORE-TIME-CLOCKSOURCE                                                         PASS                 1.03 
[hyperv]     7 BVT-CORE-TIME-CLOCKEVENT-UP                                                       PASS                 1.06 
[hyperv]     8 BVT-NET-IFUP-IFDOWN                                                               PASS                 5.03 
[hyperv]     9 BVT-VERIFY-DEPLOYMENT-PROVISION                                                   PASS                 2.69 
[hyperv] 	FirstBoot : PASS 
[hyperv] 	FirstBoot : Call Trace Verification : PASS 
[hyperv] 	Reboot : PASS 
[hyperv] 	Reboot : Call Trace Verification : PASS
```
